app/actions/websocket/group.test.ts

```diff
@@ -0,0 +1,280 @@
+
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            expect(deleteGroupChannelById).toHaveBeenCalledWith(database, 'association-id');
+        });
+    });
+
+    describe('error handling', () => {
+        it('should handle errors and fetch groups on failure', async () => {
+            const msg = {
+                broadcast: {
+                    team_id: teamId,
+                    channel_id: channelId,
+                    user_id: userId,
+                },
+                data: {
+                    group: 'invalid json',
+                },
+            } as WebSocketMessage;
+
+            await handleGroupReceivedEvent(serverUrl, msg);
+
+            expect(fetchGroupsForTeam).toHaveBeenCalledWith(serverUrl, teamId);
+            expect(fetchGroupsForChannel).toHaveBeenCalledWith(serverUrl, channelId);
+            expect(fetchGroupsForMember).toHaveBeenCalledWith(serverUrl, userId);
+        });
+
+        it('should handle invalid json on all events', async () => {
+            const msg = {
+                broadcast: {
+                    team_id: teamId,
+                },
+                data: {
+                    group: 'invalid json',
+                    group_member: 'invalid json',
+                    group_team: 'invalid json',
+                    group_channel: 'invalid json',
+                },
+            } as WebSocketMessage;
+
+            await handleGroupReceivedEvent(serverUrl, msg);
+            await handleGroupMemberAddEvent(serverUrl, msg);
+            await handleGroupMemberDeleteEvent(serverUrl, msg);
+            await handleGroupTeamAssociatedEvent(serverUrl, msg);
+            await handleGroupTeamDissociateEvent(serverUrl, msg);
+            await handleGroupChannelAssociatedEvent(serverUrl, msg);
+            await handleGroupChannelDissociateEvent(serverUrl, msg);
+
+            expect(fetchGroupsForTeam).toHaveBeenCalledWith(serverUrl, teamId);
+            expect(fetchGroupsForTeam).toHaveBeenCalledTimes(7);
+        });
+
+        it('should handle missing data', async () => {
+            const msg = {
+                broadcast: {
+                    team_id: teamId,
+                },
+                data: {},
+            } as WebSocketMessage;
+
+            await handleGroupReceivedEvent(serverUrl, msg);
+            await handleGroupMemberAddEvent(serverUrl, msg);
+            await handleGroupMemberDeleteEvent(serverUrl, msg);
+            await handleGroupTeamAssociatedEvent(serverUrl, msg);
+            await handleGroupTeamDissociateEvent(serverUrl, msg);
+            await handleGroupChannelAssociatedEvent(serverUrl, msg);
+            await handleGroupChannelDissociateEvent(serverUrl, msg);
+
+            expect(DatabaseManager.getServerDatabaseAndOperator).not.toHaveBeenCalled();
+        });
+
+        it('should handle missing database operator', async () => {
```

Comment:
does this test make sense here?

Reply:
It is a bit duplicative, I'll remove it.